builtin/game/chat.lua

```diff
@@ -1,27 +1,40 @@
 -- Minetest: builtin/game/chat.lua
 
+-- Helper function made global for use in mods
+function core.safe_gsub(s, replace, with)
+	local i1, i2 = s:find(replace, 1, true)
+	if not i1 then
+		return s
+	end
+
+	return s:sub(1, i1 - 1) .. with .. s:sub(i2 + 1)
+end
+
 --
 -- Chat message formatter
 --
 
 -- Implemented in Lua to allow redefinition
 function core.format_chat_message(name, message)
-	local str = core.settings:get("chat_message_format")
 	local error_str = "Invalid chat message format - missing %s"
-	local i
+	local str = core.settings:get("chat_message_format")
+	local old_str = str
 
-	str, i = str:gsub("@name", name, 1)
-	if i == 0 then
+	str = core.safe_gsub(old_str, "@name", name)
+	if old_str == str then
```

Comment:
surely it would have been easier to extend `safe_gsub` to return 0 or 1 instead of doing this string comparison stuff

Reply:
Yes, indeed...

Reply:
Done